frappe/email/doctype/email_account/email_account.js

```diff
@@ -68,15 +68,19 @@ frappe.email_defaults_pop = {
 
 function oauth_access(frm) {
```

Comment:
this nested callback needs to improve.

Reply:
If we don't want to make another whitelisted method outside of the controller class, I think you can use `with_doc` and `get_doc` to do it.

```javascript
function oauth_access(frm) {
	frappe.model.with_doc("Connected App", frm.doc.connected_app, () => {
		const connected_app = frappe.get_doc("Connected App", frm.doc.connected_app);
		return frappe.call({
			doc: connected_app,
			method: "initiate_web_application_flow",
			callback: function(r) {
				window.open(r.message, "_self");
			}
		});
	})
}
```